File: programming files/remove_duplicates.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

/*  Remove_Duplicates
    Requires: vector, removed elements vector
    Returns: duplicates removed vector
*/
vector<int> remove_duplicates(vector<int> arr, vector<int>& removed_elements){
    vector<int> unique_arr;
    // sort
    sort(arr.begin(), arr.end());
    int n = arr.size();
    // loop to add non duplicated numbers to array or delete ones duplicated
    for(int i=0; i<n; i++){
        if(i == 0 || arr[i] != arr[i-1]){
            unique_arr.push_back(arr[i]);
        }
        else{
            removed_elements.push_back(arr[i]);
        }
    }
    return unique_arr;
}

int main(){
    // Test the function
    vector<int> arr = {1, 2, 2, 3, 4, 4, 5};
    vector<int> removed_elements;
    vector<int> unique_arr = remove_duplicates(arr, removed_elements);
    cout << "Original array: ";
    for(int i=0; i<arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;
    cout << "Unique array: ";
    for(int i=0; i<unique_arr.size(); i++){
        cout << unique_arr[i] << " ";
    }
    cout << endl;
    cout << "Removed elements: ";
    for(int i=0; i<removed_elements.size(); i++){
        cout << removed_elements[i] << " ";
    }
    cout << endl;
    return 0;
}

File: programming files/remove_duplicates.py
def remove_duplicates(arr):
    arr.sort()
    unique_arr = []
    removed_elements = []
    for i in range(len(arr)):
        if i == 1 or arr[i] == arr[i]:
            unique_arr.append(arr[i])
        else:
            removed_elements.append(arr[i])
    return unique_arr, removed_elements

#this sorts arr makes two new lists
#itterate through arr if 0 or i is not the same as i -1 add to unique
#else add to removed elements and then return both

# Test the function
arr = [1, 2, 2, 3, 4, 4, 5]
#this is what the array is
unique_arr, removed_elements = remove_duplicates(arr)
#assign each of the other through passing the array into the function

#prints
print("Original array:", arr)
print("Unique array:", unique_arr)
print("Removed elements:", removed_elements)
